client/image_push.go

```diff
@@ -36,6 +38,20 @@ func (cli *Client) ImagePush(ctx context.Context, image string, options image.Pu
 		}
 	}
 
+	if options.Platform != nil {
+		if err := cli.NewVersionError(ctx, "1.46", "platform"); err != nil {
+			return nil, err
+		}
+
+		p := *options.Platform
+		pJson, err := json.Marshal(p)
+		if err != nil {
+			return nil, fmt.Errorf("invalid platform: %v", err)
+		}
+
+		query.Set("platform", string(pJson))
```

Comment:
Not for this PR 👈 , but if we decide to use the JSON format in a wider scope (instead of the string representation), we should consider adding a utility, similar to what we did for EncodeAuthConfig. Yes, the code is fairly trivial, but if would allow returning a proper error-type, as well as have a central location where we could define our "rules" (which could even include "reject fields XYZ" on the client side).